File: src/main/java/com/yai/YaInterpreter.java
package com.yai;

import com.yai.evaluator.StatementEvaluator;
import com.yai.model.Value;
import org.antlr.v4.runtime.BailErrorStrategy;
import org.antlr.v4.runtime.CharStreams;
import org.antlr.v4.runtime.CommonTokenStream;
import org.antlr.v4.runtime.tree.ParseTree;

import java.util.Map;


public class YaInterpreter {

	public Map<String, Value> evaluate(String statement) {
        YaInterpreterLexer lexer = new YaInterpreterLexer(CharStreams.fromString(statement));
		CommonTokenStream tokens = new CommonTokenStream(lexer);
		YaInterpreterParser parser = new YaInterpreterParser(tokens);
		parser.setErrorHandler(new BailErrorStrategy());
		ParseTree tree = parser.statement();
        System.out.println(tree.toStringTree(parser));
		return new StatementEvaluator(lexer, tree).getVariables();
	}
}

File: src/main/kotlin/com/yai/YaInterpreter.kt
//package com.yai
//
//import org.antlr.v4.runtime.BailErrorStrategy
//import org.antlr.v4.runtime.CharStreams
//import org.antlr.v4.runtime.CommonTokenStream
//import org.antlr.v4.runtime.tree.ParseTree
//
//class YaInterpreter {
//
//    fun evaluate(statement: String?): Map<String, Double> {
//        val lexer = YaInterpreterLexer(CharStreams.fromString(statement))
//        val tokens = CommonTokenStream(lexer)
//        val parser = YaInterpreterParser(tokens)
//        parser.errorHandler = BailErrorStrategy()
//        val tree: ParseTree = parser.statement() as ParseTree
//        return StatementEvaluator(lexer, tree).getVariables()
//    }
//}